aws-samples/amazon-sumerian-hosts
# Implement GitHub Actions for GitHub Pages for our docs

## Description
With the change we've made to remove Docs from the commit process, we need to update our GitHub Pages to accurately reflect how they will publish. This change is meant to bring that up-to-date to ensure it works with the new `mainline2.0` changes, and adheres to Amazon's Inclusive Language initiative. 

This change targets `mainline` intentionally, and will likely fail its first run, since mainline also has a capitalization of "README" (see changes below). This action should not fire more than the initial push as subsequent changes to `mainline` will not trigger it until `mainline2.0` is merged. Once `mainline2.0` is merged on top of `mainline`, this should trigger after every push as expected.

Finally, with this change, once we push `mainline2.0` onto `mainline`, we will still need to modify the GitHub Pages settings that we have to point to the new branch: `gh-pages` at `/(root)`. We can then safely delete the `master` branch

### Changes
* Reworked the existing script to push to "gh-pages" rather than "master." This also allows us to continue to use mainline and the current GitHub Pages until we are ready to move on
* Switched from [ad-m's GitHub Push action](https://github.com/ad-m/github-push-action) to [peaceiris's GitHub Pages Action ](https://github.com/peaceiris/actions-gh-pages), which allows us to more finely specify what we want pushed to the branch (instead of just pushing the entire existing repo)
* Fixed outstanding issues preventing `npm run docs` from running in this action
  * "readme" is case-sensitive when looked from this action (this did not occur as a problem locally), so README for the docs was changed
  * Several comments from Babylon's HostObject's file were breaking the docs from compiling. A few of there were due to `'` in the comments, with one not having any information on the return

## Reviewer Testing Instructions
* Pull PR changes to local
* Think of an upstream branch in your fork you wish to push to
* Change the `mainline` branch in the code to the upstream branch name
* Commit change
* Create an upstream branch with PR changes
* Verify GitHub actions succeeds and "gh-pages" branch is created and looks similar to the ones found from "Testing done" 

## Testing done
* Successful run from `mainline2.0`: https://github.com/brhook-aws/amazon-sumerian-hosts/actions/runs/2198204089
* Failed run from `mainline`: https://github.com/brhook-aws/amazon-sumerian-hosts/actions/runs/2198230652
  * Success after changing "README" to "readme" in `mainline` : https://github.com/brhook-aws/amazon-sumerian-hosts/actions/runs/2198241109
* Verified `gh-pages` branch looked correct:
  *  `mainline2.0`: https://github.com/brhook-aws/amazon-sumerian-hosts/tree/gh-pages-2
  * `mainline`: https://github.com/brhook-aws/amazon-sumerian-hosts/tree/gh-pages
* Verified GitHub Pages looked correct: 
  * `mainline2.0`: https://brhook-aws.github.io/amazon-sumerian-hosts/ 
  * `mainline` was not manually verified

---
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
